intercept/src/intercept.cpp

```diff
@@ -7244,13 +7304,212 @@ void CLIntercept::setKernelArgUSMPointer(
+    if (byKernelName)
+        fileNamePrefix += getShortKernelName(kernel);
+    else
+        fileNamePrefix += std::to_string(enqueueCounter);
+    fileNamePrefix += "/";
+    OS().MakeDumpDirectories( fileNamePrefix );
+    std::ofstream output{fileNamePrefix + "worksizes.txt"};
+
+    // Print the values of the worksizes and offsets on a line in the order:
+    // gws
+    // lws
+    // gws_offset
+    for (unsigned idx = 0; idx != work_dim; ++idx)
+    {
+        output << (!gws        ? 0 : gws[idx]) << ' ';
+    }
+    output << '\n';
+
+    for (unsigned idx = 0; idx != work_dim; ++idx)
+    {
+        output << (!lws        ? 0 : lws[idx]) << ' ';
+    }
+    output << '\n';
+
+    for (unsigned idx = 0; idx != work_dim; ++idx)
+    {
+        output << (!gws_offset ? 0 : gws_offset[idx]) << ' ';
+    }
+    output << '\n';
+
+    // Get the cl_program from the cl_kernel and the device(s), then extract kernel build options from the cl_program
+    CLIntercept* pIntercept = GetIntercept();
+
+    cl_program tmp_program;
+    pIntercept->dispatch().clGetKernelInfo(kernel, CL_KERNEL_PROGRAM, sizeof(cl_program), &tmp_program, nullptr);
+
+    cl_context context;
+    pIntercept->dispatch().clGetProgramInfo(tmp_program, CL_PROGRAM_CONTEXT, sizeof(cl_context), &context, nullptr);
+
+    cl_device_id device_ids;
+    pIntercept->dispatch().clGetContextInfo(context, CL_CONTEXT_DEVICES, sizeof(cl_context_info*), &device_ids, nullptr);
+
+    size_t sizeOfOptions = 0;
+    pIntercept->dispatch().clGetProgramBuildInfo(tmp_program, device_ids,
+                                                 CL_PROGRAM_BUILD_OPTIONS, sizeof(char*), nullptr, &sizeOfOptions);
+    char* optionsString = new char[sizeOfOptions];
+    pIntercept->dispatch().clGetProgramBuildInfo(tmp_program, device_ids,
+                                                 CL_PROGRAM_BUILD_OPTIONS, sizeOfOptions, optionsString, &sizeOfOptions);
+
+    std::ofstream outputBuildOptions{fileNamePrefix + "buildOptions.txt"};
+    outputBuildOptions << optionsString;
+    outputBuildOptions << '\n';
+    delete[] optionsString;
+
+    std::string knlName = getShortKernelName(kernel);
+    std::ofstream outputKnlName{fileNamePrefix + "knlName.txt"};
+    outputKnlName << knlName;
+
+    std::ofstream outputPythonScript{fileNamePrefix + "run.py"};
+    outputPythonScript << pythonScript << '\n';
+
+    std::ofstream outputKernelNumber{fileNamePrefix + "enqueueNumber.txt"};
+    outputKernelNumber << std::to_string(enqueueCounter) << '\n';
+
+    cl_uint numArgs = 0;
+    pIntercept->dispatch().clGetKernelInfo(kernel, CL_KERNEL_NUM_ARGS, sizeof(cl_uint), &numArgs, nullptr);
+
+    std::ofstream outputArgTypes{fileNamePrefix + "ArgumentDataTypes.txt"};
+    for ( int idx = 0; idx != numArgs; ++idx )
+    {
+        char tmp[100];
+        int error = pIntercept->dispatch().clGetKernelArgInfo(kernel, idx, CL_KERNEL_ARG_TYPE_NAME, sizeof(tmp), tmp, nullptr);
```

Comment:
We shouldn't assume that the maximum kernel arg type name is 100 bytes (especially for C++ programs with template expansions!).  It would be better to read this into a std::vector or a std::string instead.